[stable24] Merge pull request #1273 from nextcloud/enh/open_with

Raudius opened the pull request:
Manual backport for: https://github.com/nextcloud/viewer/pull/1273

Add `openWith` function to OCA.Viewer

juliushaertl commented:
@Raudius Just noticed, It would be nice to have the commits being cherry-picked instead of the merge commit, so we preserve the original commit details. :)

Raudius commented:
@juliushaertl 

Fixed the commits. Also fixed the pipeline, but I had to do an `npm update` which messed results in a big diff on the .lock, do you know why this might be?

juliushaertl commented:
> Fixed the commits. Also fixed the pipeline, but I had to do an npm update which messed results in a big diff on the .lock, do you know why this might be?

npm update shouldn't be required. `npm ci` would install the dependencies on the current branch based on what is stated in the package-lock.json file. Do you remember what lead to running npm update?
